clients/cobol-lsp-vscode-extension/src/services/LanguageClientService.ts

```diff
@@ -71,31 +67,26 @@ export class LanguageClientService {
 
     public async retrieveAnalysis(uri: string, text: string): Promise<any> {
         const languageClient = this.getLanguageClient();
-        await languageClient.onReady();
         return languageClient.sendRequest("extended/analysis", { uri, text });
     }
 
     public async invalidateConfiguration() {
         const languageClient = this.getLanguageClient();
-        await languageClient.onReady();
         languageClient.sendNotification(DidChangeConfigurationNotification.type, { settings: null });
     }
 
-    public start(): vscode.Disposable {
-        this.initHandlers();
-        return this.getLanguageClient().start();
+    public async start() {
```

Comment:
Why it's not disposable anymore?

Reply:
Yes. Both start and stop methods now return a promise.

ref - https://github.com/microsoft/vscode-languageserver-node#3172-protocol-802-json-rpc-802-client-and-802-server